app/src/main/java/pm/gnosis/heimdall/ui/safe/create/CreateSafeConfirmRecoveryPhraseViewModel.kt

```diff
@@ -50,6 +68,7 @@ class CreateSafeConfirmRecoveryPhraseViewModel @Inject constructor(
             .flatMap { request -> relayServiceApi.safeCreation(request).map { request to it } }
             .flatMap { (request, response) ->
                 assertResponse(request, response)
+                gnosisSafeRepository.assignOwnerToSafe(request.owners[0], response.safe)
```

Comment:
I would probably not create the owner in the db until this point.

This would also prevent that we create owner entries without safes (which might occur if we have a network error or the user closes the app).

Also this would allow us to make the safe a primary key in the table

Reply:
yes, good point